### PR TITLE
[Static reference] LPS-170478 Remove method reference by using snapshot in modules calendar-web, journal-web

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:calendar

**Comment by zhangyan0701** [comment]
ci:test:echo

**Comment by W2jx** [comment]
no conflicts(4/7 16:38)

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

**Comment by W2jx** [comment]
no conflicts(4/10 8:33)

**Comment by W2jx** [comment]
https://github.com/liferay-core-infra/liferay-portal/pull/4092

**Comment by W2jx** [comment]
no conflicts(4/11 10:10)

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/util/CalendarUtil.java [hunk 1]
```diff
@@ -168,6 +168,9 @@ public static JSONObject toCalendarBookingJSONObject(
 			TimeZone timeZone)
 		throws PortalException {
 
+		CalendarResourceLocalService calendarResourceLocalService =
```
Comment (dantewang) [inline modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/util/CalendarUtil.java:171]: @W2jx Please put this one in the lambda.

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/util/CalendarUtil.java [hunk 2]
```diff
@@ -202,11 +205,14 @@ public static JSONObject toCalendarBookingJSONObject(
...
 
+		WorkflowInstanceLinkLocalService workflowInstanceLinkLocalService =
+			_workflowInstanceLinkLocalServiceSnapshot.get();
+
 		jsonObject.put(
 			"hasChildCalendarBookings", childCalendarBookings.size() > 1
 		).put(
 			"hasWorkflowInstanceLink",
-			_workflowInstanceLinkLocalService.hasWorkflowInstanceLink(
+			workflowInstanceLinkLocalService.hasWorkflowInstanceLink(
```
Comment (dantewang) [inline modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/util/CalendarUtil.java:215]: @W2jx Please use `JSONObject.put(key, supplier)` and get the service in the supplier whenever fits the usage for the variables you replaced to make the long chain clearer.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**